config/config.go

```diff
@@ -78,18 +85,30 @@ func New() *Configuration {
 	return c
 }
 
-// NewFromValues returns a new *config.Configuration instance as if it had
-// been read from the .gitconfig specified by "gitconfig" parameter.
+// Values is a convenience type used to call the NewFromValues function. It
+// specifies `Git` and `Env` maps to use as mock values, instead of calling out
+// to real `.gitconfig`s and the `os.Getenv` function.
+type Values struct {
+	// Git and Env are the stand-in maps used to provide values for their
+	// respective environments.
+	Git, Env map[string]string
+}
+
+// NewFrom returns a new `*config.Configuration` that reads both its Git
+// and Enviornment-level values from the ones provided instead of the actual
+// `.gitconfig` file or `os.Getenv`, respectively.
 //
-// NOTE: this method should only be called during testing.
-func NewFromValues(gitconfig map[string]string) *Configuration {
+// This method should only be used during testing.
+func NewFrom(v Values) *Configuration {
 	config := &Configuration{
+		Env: mapFetcher(v.Env),
```

Comment:
We shouldn't need to synchronize access to this map, since all the data is provided up front in #1416 and all subsequent map operations are read-only. No writers and `n` readers will not result in a concurrent map access error.